### DISCUSSION TITLE
MRTK Ray Interactor Visual with Layer Filter

### DISCUSSION THREAD

**Discussion by mmaurer42612** [microsoft/MixedRealityToolkit-Unity#10945] [Ideas]
Hi, i am currently using the MRTK3 UX Components and the MRTK Ray Interactor for the Input (mainly with XR Controller Input). I want to ask if there is the possibility to only show the interaction ray when hovering over an UI-Element (They are all on the UI Layer in my case). I thought this can be done with the Raycast Mask on the Ray Interactor, but the Ray Interactor Visual kind of seems to interrupt that. Is there another way to achieved the preferred behaviour or could something like this be implemented in the Ray Interactor Visual?

Thanks in advance, 
Martina

**Comment by Zee2** [comment]
Hi @mmaurer42612 , which version are you on? I believe `pre.10` included some modifications to the ray visual surrounding UI hit tests.

**Reply by mmaurer42612** [reply]
I was on version pre.9, with pre.10 achieved what i wanted to do, thanks a lot!
